### ISSUE TITLE
Apple2: .dsk-based images mounted as "read-only" can be written to.

### ISSUE THREAD

**Issue by Shoegzer** [mamedev/mame#8476]
Images mounted at least as "dsk" (not via softlist) as "read-only" via MAME's internal UI can be written to via apple2 disk IO.  

Test process: 

* Launch MAME without software
* Mount DOS .dsk image as "Read-only" in UI
* Select "Reset" in UI to hard-reset system
* After bootup, write small Applesoft BASIC program \<PROG> and type "SAVE \<PROG>"  
* Type CATALOG and note existence of new file \<PROG> 

Information:

* CPU: Intel i7-10870H
* OS: Linux Mint 20.2 / kernel 5.11
* Compiler: gcc 11.1.0
* MAME Version: 0.234

**Comment by cuavas** [comment]
Does it get written back to the disk image on exit though?  This may be a case where “read-only“ only applies to image file handling (i.e. changes are ephemeral).

**Comment by peterferrie** [comment]
> Does it get written back to the disk image on exit though?  This may be a
> case where “read-only“ only applies to image file handling (i.e. changes are
> ephemeral).

Writes to .dsk appear to succeed but are discarded on exit, unlike
.woz images which are actually written (and currently damaged).

**Comment by Shoegzer** [comment]
@cuavas It does indeed write back on exit (i.e. writes to the .dsk persist after restarting MAME).  I just tested this again myself.
@peterferrie I have not been able to replicate a situation where the writes don't succeed.  Did you observe this according to my instructions above?

**Comment by peterferrie** [comment]
> @cuavas It does indeed write back on exit (i.e. writes to the .dsk persist
> after restarting MAME).  I just tested this again myself.
> @peterferrie I have not been able to replicate a situation where the writes
> don't succeed.  Did you observe this according to my instructions above?

I can't get it to write to the image unless it's not read-only,
following exactly your instructions, other than I'm using Windows
rather than Linux.
Perhaps an OS-level difference.

**Comment by Shoegzer** [comment]
Agreed, could indeed be OS-level.  Would be great to have confirmation from someone else with linux.
